parquet/src/test/java/org/apache/iceberg/parquet/TestCDHParquetStatistics.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.parquet;
+
+import org.apache.parquet.column.statistics.Statistics;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for Parquet 1.5.0-Stats which cannot be evaluated like later versions of Parquet stats. They are intercepted
+ * by the hasNonNullButNoMinMax function which always returns ROWS_MAY_MATCH
+ */
+public class TestCDHParquetStatistics {
+
+  @Test
+  public void testCDHParquetStatistcs() {
+    Statistics cdhBinaryColumnStats = mock(Statistics.class);
+    when(cdhBinaryColumnStats.getMaxBytes()).thenReturn(null);
+    when(cdhBinaryColumnStats.getMinBytes()).thenReturn(null);
+    when(cdhBinaryColumnStats.getNumNulls()).thenReturn(0L);
+    Assert.assertTrue(ParquetMetricsRowGroupFilter.hasNonNullButNoMinMax(cdhBinaryColumnStats, 50L));
```

Comment:
Should we also test -1 for num nulls? I know we don't expect it, but the behavior should still be correct, I think.

Reply:
Sure I can add that, techinically since this is in the inner loop of "notEmpty" either nonNull >= 0 (set) or haveNonNulls is true. If hasNoNulls is true then we don't care about this pathway anyway because max/min are set

Reply:
Yeah, I just want to avoid having a method that assumes something about the context in which it will be called. Probably best to make sure it behaves correctly in all situations. Looks like if num nulls is unknown, -1, this will return that there are non-null values, even if the value count is 0. We might want to fix that.

Reply:
I can add that, but in the usage again this is past the "valueCount ==0" which causes no match in the first return